model/RepositoryService.php

```diff
@@ -94,7 +94,7 @@ public function commit($resourceId, $message, $version = null, $userId = null)
     {
         if ($userId === null) {
             $user = \common_session_SessionManager::getSession()->getUser();
-            $userId = is_null($user) ? null : $user->getIdentifier();
+            $userId = is_null($user) ? '' : $user->getIdentifier();
```

Comment:
While updating this we might as well change `is_null($user)` to the more appropriate `$user === null`